### PR TITLE
Add field for project objective and load offer content

### PR DESCRIPTION
**Pull request by sven1103**

### CODE REVIEW

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/project/create/CreateProjectLayout.java [hunk 1]
```diff
@@ -47,6 +50,10 @@ protected VerticalLayout initContent() {
 
     FormLayout formLayout = new FormLayout();
     formLayout.addFormItem(titleField, "Project Title");
+    formLayout.addFormItem(projectObjective, "Project Objective");
+    formLayout.setResponsiveSteps(new ResponsiveStep("200px", 1));
```
Comment (KochTobi) [inline vaadinfrontend/src/main/java/life/qbic/datamanager/views/project/create/CreateProjectLayout.java:54]: As I just tackled the same issue just a short finding:
The min width is where you want the layout to behave like having the column count you specify.
As we (currently) only want one column independent of width, I would suggest to change this line
```suggestion
    formLayout.setResponsiveSteps(new ResponsiveStep("0", 1));
```
reference: https://vaadin.com/docs/latest/components/form-layout#sectioning

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/project/create/CreateProjectLayout.java [hunk 2]
```diff
@@ -32,6 +33,8 @@ public class CreateProjectLayout extends Composite<VerticalLayout> implements Ha
   final Button saveButton = new Button("Save");
   final Button cancelButton = new Button("Cancel");
 
+  final TextField projectObjective = new TextField();
```
Comment (jenniferboedker) [inline vaadinfrontend/src/main/java/life/qbic/datamanager/views/project/create/CreateProjectLayout.java:36]:
```suggestion
  final TextArea projectObjective = new TextArea();
```

File: projectmanagement/src/main/java/life/qbic/projectmanagement/domain/project/ProjectIntent.java [hunk 3]
```diff
@@ -39,12 +47,11 @@ public boolean equals(Object o) {
...
     ProjectIntent that = (ProjectIntent) o;
 
-    return projectTitle.equals(that.projectTitle);
+    return projectTitle.equals(that.projectTitle) && projectObjective.equals(that.projectObjective);
   }
 
   @Override
   public int hashCode() {
-    return projectTitle.hashCode();
+    return projectTitle.hashCode(); //todo integrate objective hash -> du it with project intent?
```
Comment (jenniferboedker) [inline projectmanagement/src/main/java/life/qbic/projectmanagement/domain/project/ProjectIntent.java:55]: Maybe @KochTobi can tell me on how to calculate the hashcode for multiple values?

File: projectmanagement/src/main/java/life/qbic/projectmanagement/domain/project/ProjectIntent.java [hunk 4]
```diff
@@ -39,12 +47,11 @@ public boolean equals(Object o) {
 
     ProjectIntent that = (ProjectIntent) o;
 
-    return projectTitle.equals(that.projectTitle);
+    return projectTitle.equals(that.projectTitle) && projectObjective.equals(that.projectObjective);
```
Comment (jenniferboedker) [inline projectmanagement/src/main/java/life/qbic/projectmanagement/domain/project/ProjectIntent.java:50]: How do we need to calculate this mehtod according to ddd? @KochTobi

File: projectmanagement/src/main/java/life/qbic/projectmanagement/domain/project/ProjectIntent.java (hunk 3 above)
Comment (KochTobi) [reply projectmanagement/src/main/java/life/qbic/projectmanagement/domain/project/ProjectIntent.java:55]: Ask IntelliJ.
Simply press command + N and choose generate equals and hash code

File: projectmanagement/src/main/java/life/qbic/projectmanagement/domain/project/ProjectIntent.java (hunk 4 above)
Comment (KochTobi) [reply projectmanagement/src/main/java/life/qbic/projectmanagement/domain/project/ProjectIntent.java:50]: yes you can remove the equals and hash code methods and regenerate them.
Be sure that all fields are included and marked as non-null

File: projectmanagement/src/main/java/life/qbic/projectmanagement/application/ProjectCreationService.java [hunk 5]
```diff
@@ -32,10 +29,13 @@ public ProjectCreationService(ProjectRepository projectRepository) {
    * @param title the title of the project.
    * @return the created project
    */
-  public Result<Project, ApplicationException> createProject(String title) {
+  public Result<Project, ApplicationException> createProject(String title, String objective) {
     ProjectTitle projectTitle;
+    ProjectObjective projectObjective;
+    //add exp. design
```
Comment (KochTobi) [inline projectmanagement/src/main/java/life/qbic/projectmanagement/application/ProjectCreationService.java:35]:
```suggestion
```
has nothing to do with this branch

File: projectmanagement/src/main/java/life/qbic/projectmanagement/domain/project/ProjectObjective.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+import java.util.Objects;
+
+/**
+ * <b>short description</b>
+ *
+ * <p>detailed description</p>
+ *
+ * @since <version tag>
+ */
```
Comment (KochTobi) [inline projectmanagement/src/main/java/life/qbic/projectmanagement/domain/project/ProjectObjective.java:11]: Please provide some javadoc for the domain concept of a project objective

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/project/create/CreateProjectHandler.java [hunk 7]
```diff
@@ -71,7 +71,10 @@ private void addSaveClickListener() {
 
   private void saveClicked() {
     String titleFieldValue = createProjectLayout.titleField.getValue();
-    projectCreationService.createProject(titleFieldValue)
+    String objectiveFieldValue = createProjectLayout.projectObjective.getValue();
+    //add exp design
+
```
Comment (KochTobi) [inline vaadinfrontend/src/main/java/life/qbic/datamanager/views/project/create/CreateProjectHandler.java:76]:
```suggestion
    //add exp design

```

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/project/create/CreateProjectLayout.java [hunk 8]
```diff
@@ -47,6 +51,10 @@ protected VerticalLayout initContent() {
 
     FormLayout formLayout = new FormLayout();
     formLayout.addFormItem(titleField, "Project Title");
+    formLayout.addFormItem(projectObjective, "Project Objective");
+    formLayout.setResponsiveSteps(new ResponsiveStep("0", 1));
+    titleField.setSizeFull();
+
```
Comment (KochTobi) [inline vaadinfrontend/src/main/java/life/qbic/datamanager/views/project/create/CreateProjectLayout.java:57]:
```suggestion
    projectObjective.setWidthFull();
```
Do you want to increase the size of the objective as well?

File: projectmanagement/src/main/java/life/qbic/projectmanagement/application/ProjectCreationService.java [hunk 9]
```diff
@@ -47,15 +46,15 @@ public Result<Project, ApplicationException> createProject(String title) {
     }
     Project project;
     try {
-      project = Project.create(new ProjectIntent(projectTitle));
+      project = Project.create(new ProjectIntent(projectTitle,projectObjective));
```
Comment (KochTobi) [inline projectmanagement/src/main/java/life/qbic/projectmanagement/application/ProjectCreationService.java:49]:
```suggestion
      project = Project.create(new ProjectIntent(projectTitle, projectObjective));
```

**Pull request merged**